src/go/k8s/pkg/resources/configmap.go

```diff
@@ -103,17 +172,30 @@ func (r *ConfigMapResource) Obj() (k8sclient.Object, error) {
 		return nil, err
 	}
 
+	kafkaInternalPort, kafkaExternalPort := r.getKafkaAdvertisedPorts()
+
 	script :=
 		`set -xe;
 		CONFIG=` + configPath + `;
 		ORDINAL_INDEX=${HOSTNAME##*-};
 		SERVICE_NAME=${HOSTNAME}.` + serviceAddress + `
-		cp /mnt/operator/redpanda.yaml $CONFIG;
+		cp ` + filepath.Join(configuratorDir, confFile) + ` $CONFIG;
 		rpk --config $CONFIG config set redpanda.node_id $ORDINAL_INDEX;
 		if [ "$ORDINAL_INDEX" = "0" ]; then
 			rpk --config $CONFIG config set redpanda.seed_servers '[]' --format yaml;
-		fi;
-		cat $CONFIG`
+		fi;`
+	var endScript string
+	if r.pandaCluster.Spec.ExternalConnectivity {
+		endScript = `
+			cat ` + nodeListPath + `
+			EXTERNAL_IP=$(cat ` + nodeListPath + ` | grep $HOST_IP | awk '{print $2}')
+			rpk --config $CONFIG config set redpanda.advertised_kafka_api --format json '[{"name": "external","address":"'$EXTERNAL_IP'","port":` + kafkaExternalPort + `},{"name": "internal","address":"'$SERVICE_NAME'","port":` + kafkaInternalPort + `}]'
```

Comment:
why can't this be passed to redpanda on rpk start as part of the config?

Reply:
Because on the line above we `cat | grep | awk` the `ExternalIP` variable which is unavailable in the Kubernetes downward API. 

More context:
It's a chicken and an egg problem. We need to know the IP address of the node on which the POD will be scheduled. We would like to live the scheduling to the Kuberntes. There is another problem with statefulset which need unified declaration of the pod spec across all replicas. There is no other time in the lifecycle to retrieve dynamically the node IP information other than in the init container.

Reply:
From the docs:

> The following information is available through environment variables:
> * status.podIP - the pod's IP address
> * spec.serviceAccountName - the pod's service account name, available since v1.4.0-alpha.3
> * spec.nodeName - the node's name, available since v1.4.0-alpha.3
> * status.hostIP - the node's IP, available since v1.7.0-alpha.1

REF https://kubernetes.io/docs/tasks/inject-data-application/downward-api-volume-expose-pod-information/

Unfortunately hostIP is the private IP of the host.

To verify I run the following test.

```
# cat dapi-envars-pod.yaml
apiVersion: v1
kind: Pod
metadata:
  name: dapi-envars-fieldref
spec:
  containers:
    - name: test-container
      image: k8s.gcr.io/busybox
      command: [ "sh", "-c"]
      args:
      - while true; do
          echo -en '\n';
          printenv MY_HOST_IP;
          sleep 10;
        done;
      env:
        - name: MY_HOST_IP
          valueFrom:
            fieldRef:
              fieldPath: status.hostIP
  restartPolicy: Never

# ekc create cluster
....
# kubectl apply -f dapi-envars-pod.yaml
pod/dapi-envars-fieldref created
# kubectl get pod -o custom-columns=NAME:.metadata.name,IP:.status.podIP,Node:.spec.nodeName
NAME                   IP              Node
dapi-envars-fieldref   192.168.31.49   ip-192-168-16-111.eu-west-2.compute.internal
# kubectl get node ip-192-168-16-111.eu-west-2.compute.internal -o custom-columns='NAME:.metadata.name,EXTERNAL-IP:.status.addresses[?(@.type=="ExternalIP")].address,INTERNAL-IP:.status.addresses[?(@.type=="InternalIP")].address'
NAME                                           EXTERNAL-IP   INTERNAL-IP
ip-192-168-16-111.eu-west-2.compute.internal   3.8.157.0     192.168.16.111
# kubectl logs dapi-envars-fieldref

192.168.16.111
```